pkg/collector/containerlogs_containerd_collector.go

```diff
@@ -0,0 +1,160 @@
+		if err != nil {
+			return err
+		}
+
+		collector.data[containerLog.containerName] = output
+	}
+
+	return nil
+}
+
+func determineContainerLogsToCollect(allContainers []ContainerLog, selectors []ContainerLogSelector) []ContainerLog {
+	var selectedContainerLogs []ContainerLog
+	for _, containerLog := range allContainers {
+		for _, selector := range selectors {
+			if doesSelectorSelectContainerLog(containerLog, selector) {
+				selectedContainerLogs = append(selectedContainerLogs, containerLog)
+			}
+		}
+	}
+	return selectedContainerLogs
+}
+
+func doesSelectorSelectContainerLog(containerLog ContainerLog, selector ContainerLogSelector) bool {
+	if selector.namespace == allNamespacesSelector {
+		return true
+	}
+
+	return containerLog.namespace == selector.namespace && strings.HasPrefix(containerLog.containerName, selector.containerNamePrefix)
+}
+
+//TODO allow the raw struct objects to be defined directly in the deployment yaml and add additional required logic to DoesSelectorSelectContainerLog
+func parseContainerLogSelectors(selectorStrings []string) []ContainerLogSelector {
+	var containerLogSelectors []ContainerLogSelector
+
+	for _, selectorString := range selectorStrings {
+		selectorStringParts := strings.Split(selectorString, "/")
+
+		if len(selectorStringParts) == 1 {
+			containerLogSelectors = append(containerLogSelectors, ContainerLogSelector{
+				namespace: selectorStringParts[0],
+			})
+		}
+		if len(selectorStringParts) == 2 {
+			containerLogSelectors = append(containerLogSelectors, ContainerLogSelector{
+				namespace:           selectorStringParts[0],
+				containerNamePrefix: selectorStringParts[1],
+			})
+		}
+	}
+
+	return containerLogSelectors
+}
+
+func parseContainerLogFilenames(directoryPath string, containerLogFilenames []string) []ContainerLog {
+
+	var containerLogs []ContainerLog
+
+	//container log files are in format PODNAME_NAMESPACE_CONTAINERNAME-CONTAINERUID.log
+	//TODO check that CONTAINERNAME and CONTAINERUID are correct terminology
+	for _, logFile := range containerLogFilenames {
+		logfileSplitOnDot := strings.Split(logFile, ".")
+		logFileSplitOnUnderscore := strings.Split(logfileSplitOnDot[0], "_")
+		containerNameWithIDSplitOnDash := strings.Split(logFileSplitOnUnderscore[2], "-")
+
+		//uid is the last value
+		indexOfUid := len(containerNameWithIDSplitOnDash) - 1
+
+		//containerName is everything except the last value, joined
+		containerName := strings.Join(containerNameWithIDSplitOnDash[0:indexOfUid], "")
+
+		containerLogs = append(containerLogs, ContainerLog{
+			podname:       logFileSplitOnUnderscore[0],
+			namespace:     logFileSplitOnUnderscore[1],
+			containerName: containerName,
+			containeruid:  containerNameWithIDSplitOnDash[indexOfUid],
+			filepath:      path.Join(directoryPath, logFile),
+		})
+	}
+
+	return containerLogs
+}
+
+func getAllContainerLogFilesThatHaveEverRunOnHost() ([]string, error) {
+	output, err := utils.RunCommandOnHost("ls", containerLogDirectory)
```

Comment:
As you still use `RunCommandOnHost` I suppose we don't want to get rid of it?
Keep it mind that this implies no possible support for other OS, bad error handling, no complex structure (since we can only retrieve strings), no streaming, etc ...

Reply:
we do want to get rid of RunCommandOnHost (or at least minimize its usage for all the good points which you bring up - I think we will want to retain the capability unless it becomes the only blocker to scoping down the permissions to e.g. allow periscope to run as non-privileged pod).

This can be refactored to use volume mount, but this required significant changes to the internal structures so I punted on it for me